packages/compass-collection/src/components/mock-data-generator-modal/script-generation-utils.ts

```diff
@@ -0,0 +1,611 @@
+export type FakerArg = string | number | boolean | { json: string };
+
+const DEFAULT_ARRAY_LENGTH = 3;
+const INDENT_SIZE = 2;
+
+export interface FieldMapping {
+  mongoType: string;
+  fakerMethod: string;
+  fakerArgs: FakerArg[];
+  probability?: number; // 0.0 - 1.0 frequency of field (defaults to 1.0)
+}
+
+// Array length configuration for different array types
+export type ArrayLengthMap = {
+  [fieldName: string]:
+    | number[] // Multi-dimensional: [2, 3, 4]
+    | ArrayObjectConfig; // Array of objects
+};
+
+export interface ArrayObjectConfig {
+  length?: number; // Length of the parent array (optional for nested object containers)
+  elements: ArrayLengthMap; // Configuration for nested arrays
+}
+
+export interface ScriptOptions {
+  documentCount: number;
+  databaseName: string;
+  collectionName: string;
+  arrayLengthMap?: ArrayLengthMap;
+}
+
+export type ScriptResult =
+  | { script: string; success: true }
+  | { error: string; success: false };
+
+type DocumentStructure = {
+  [fieldName: string]:
+    | FieldMapping // Leaf: actual data field
+    | DocumentStructure // Object: nested fields
+    | ArrayStructure; // Array: repeated elements
+};
+
+interface ArrayStructure {
+  type: 'array';
+  elementType: FieldMapping | DocumentStructure | ArrayStructure;
+}
+
+/**
+ * Entry point method: Generate the final script
+ */
+export function generateScript(
+  schema: Record<string, FieldMapping>,
+  options: ScriptOptions
+): ScriptResult {
+  try {
+    const structure = buildDocumentStructure(schema);
+
+    const documentCode = renderDocumentCode(
+      structure,
+      INDENT_SIZE * 2, // 4 spaces: 2 for function body + 2 for inside return statement
+      options.arrayLengthMap
+    );
+
+    const script = `// Mock Data Generator Script
+// Generated for collection: ${options.databaseName}.${options.collectionName}
+// Document count: ${options.documentCount}
+
+const { faker } = require('@faker-js/faker');
+
+// Connect to database
+use(${JSON.stringify(options.databaseName)});
+
+// Document generation function
+function generateDocument() {
+  return ${documentCode};
+}
+
+// Generate and insert documents
+const documents = [];
+for (let i = 0; i < ${options.documentCount}; i++) {
+  documents.push(generateDocument());
+}
+
+// Insert documents into collection
+db.getCollection(${JSON.stringify(
+      options.collectionName
+    )}).insertMany(documents);
+
+console.log(\`Successfully inserted \${documents.length} documents into ${
+      options.databaseName
+    }.${options.collectionName}\`);`;
```

Comment:
^ This still needs to be escaped, `'` is a valid character in database and collection names